### ISSUE TITLE
[Feature Request]: Override default video player settings

### ISSUE THREAD

**Issue by efb4f5ff-1298-471a-8973-3d47447115dc** [FreeTubeApp/FreeTube#2295]
### Guidelines

- [X] I have searched the [issue tracker](https://github.com/FreeTubeApp/FreeTube/issues) for a feature request that matches the one I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the feature request I want to file, without success.

### Problem Description

When https://github.com/FreeTubeApp/FreeTube/issues/2263 was opened up. I noticed how many issues were actually opened for the same request. So here we are making a big summary.

Why do we want this? Let say in the settings i have set my default video quality to 1080p. My internet provider is giving me a hard time today so while im watching a video i switch from 1080p to 360p in the video player. The next video should also run in 360p because i just have overwriting the default quality because my provider giving me a hard time. 

I could write usecases for the rest but i assume u get my point. U change it in the player for a reason.

### Proposed Solution

Player buttons from the left to right, that already do this.

- [x] Volume
- [ ] Playback speed
- [ ] Captions
- [ ] PIP
- [x] Theatre mode*
- [ ] Fullwindow (can only be implemented with autoplay because u have to go out of this mode to select another video when autoplay is disabled)
- [ ] Video Quality
- [ ] Fullscreen

* There is a setting called ```Enable Theatre Mode by Default```. When this is enabled Theatre mode will always get forced on u on the next video. If this is disabled and u switch from default view to theatre mode view in the video player. It will retain these setting.

### Alternatives Considered

I do understand that some people like it that the default settings overwrite the player settings when a next video is being played. We maybe could have 2 modes?

1. Player settings is always overwritten by default settings when another video is being player.
2. Player settings dont get overwritten and stay like u set them to be for that session.

### Issue Labels

ease of use improvement

### Additional Information

I understand that this will be a difficult and long discussion. The point of this issue is to centralize the discussion to this issue instead of 8 other issues.

Allot of discussion has already taken place [here](https://github.com/FreeTubeApp/FreeTube/pull/1281).

**Comment by anonaddict** [comment]
Makes sense to keep everything for the session, if someone is watching a video playlist in full screen odds are they want to continue watching in fullscreen instead of having to focus freetube and manually fullscreen every video. Especially not if it is shorter videos like vines (if those still exist) or music (2-4 minute long videos generally)

Considering how PiP currently works I also doubt anyone watching playlists or autoplay will benefit from having to manually close PIP every video (since it blackscreens when the video switches) and reopen a new one

I don't see what usecases there would be for automatically overriding the current settings with some defaults. If anyone does have use cases that would benefit from keeping the current functionality I'd be happy to hear about them, but as things currently stand I don't see a reason to make this a toggle instead of just replacing it with the current volume functionality

**Comment by anonaddict** [comment]
#2041 related

**Comment by anonaddict** [comment]
I'd like to request this be assigned a high(er) priority

**Comment by jasonhenriquez** [comment]
I'm working on a feature adding stylized radio button groups an `ft-icon-button` dropdown. We could use this to add radio button group toggles for these video player - changeable settings:
- Playback Rate
- Video Quality
- Theater Mode
- Fullscreen
- Fullwindow
- Picture in Picture
To start off, we can probably only display levels 3, 4, and 6 of the settings manifesto, as those are the easiest. 3 is just saving in component state, 4 is using `stateWithSideEffects` if a secondary toggle is enabled (i.e., has one particular string value) for that given feature, and 6 is how we do it normally.

**Comment by jasonhenriquez** [comment]
Simpler idea: what about a toggle or checkmark on the Watch route titled something like `Remember video settings throughout session`?

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
The default settings or if it changed?

**Comment by jasonhenriquez** [comment]
Settings scope 4 on [the manifesto](https://github.com/FreeTubeApp/FreeTube/pull/1281#issuecomment-912846345) is what I had in mind:
> Another has setting changes that persist throughout the entire session but not in different sessions (i.e., how volume works currently).

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
4 seems like the most logical route to me.

**Comment by on4r4p** [comment]
So how is the "keep the fullscreen while in playlist" feature going nowaday ?

**Comment by AedDev** [comment]
Would also love to see the "keep the fullscreen while in playlist" in upcoming releases :)

**Comment by toby63** [comment]
I think four options would be best:
1. Standard setting: keep default
2. Standard setting: keep manually set options always
3. Standard setting: keep settings in one session, then reset to default
4. Dynamic setting: add a button "keep settings", so it things are kept until you unmark the button or something similar

In addition there could also be:
4.a. Keep the Setting just for the same window or playlist etc. and then reset to defaults.

**Comment by HeavyHeart2012** [comment]
Yes, we need fullscreen while playing playlists please!

**Comment by loopernow** [comment]
> In addition there could also be: 4.a. Keep the Settings just for the same window or playlist etc. and then reset to defaults.

4.a. makes perfect sense.

**Comment by edmundlaugasson** [comment]
Using 0.22.1 beta and still the issue persists - after every video in playlist full screen is taken off in order to play next video.

**Comment by nhooey** [comment]
Should the current captions toggle state already be preserved between videos (in a playlist, or not)?

If I turn on captions with the toggle keyboard shortcut on one video, the next video that autoplays from the playlist will have captions off (where captions are off by default in my settings).

This seems like suboptimal behaviour to me.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@nhooey as seen in the issue body Captions didnt receive the proper treatment yet

**Comment by fkdp06** [comment]
I also wish the media format is being saved within the session.

So my use case is: when playing a playlist and switching the media format to "Audio only", the next song on the playlist switches the format back to video. So I am not able to listen to music with audio only mode.
